inferno-framework/inferno-core
# FI-2075: Input validation

# Summary

When a user clicks on an input, consider it modified. Any modified input that is required but is left empty enters error status. 

<img width="500" alt="Screenshot 2023-08-29 at 4 55 36 PM" src="https://github.com/inferno-framework/inferno-core/assets/11209247/862dd850-38bf-49ed-be0c-08fe6ffe09d6">


# Testing Guidance

1) Enter required inputs. Submit button should become active.
2) Click into a required input that has no default value. Click away without entering a value. The input should become red. Entering a value should remove the red color. 
3) Click into a required input that has a default value and click out of it. Nothing should change.
4) Click into a required input that has a default value and clear the value. The input should become red.

Make sure to test on the following input types: text field, text area, checkbox, Oauth credentials.
Radio inputs should always have a value selected and should be impossible to empty.
